docs/design-notes/protocols.md

```diff
@@ -93,19 +93,56 @@ protocol. Scylla supports several Scylla-only extensions to this protocol,
 described in [protocol-extensions.md](protocol-extensions.md).
 
 By default, Scylla listens to the CQL protocol on port 9042, which can be
-configured via the `native_transport_port` configuration option.
+configured via the `native_transport_port` configuration option. If set
+to 0, the OS will choose the port randomly (which also applies to every CQL
+port discussed later). To explicitly disable listening on a CQL port one
+should set its value to `~` , or `null`.
 
 Scylla also supports the CQL protocol via TLS/SSL encryption which is
 disabled by default and can be enabled via the `native_transport_port_ssl`
-configuration option (the traditional choice is port 9142). Users
-that want to only support the encrypted CQL protocol can disable the
-unencrypted default CQL protocol by setting the `native_transport_port` to 0.
+configuration option. Traditionally, the secure connections are accepted
+on port 9142, but if `client_encryption_options` are specified and
+`native_transport_port_ssl` is not, then `native_transport_port` will
+handle encrypted connections only. The same thing happens when
+`native_transport_port` and `native_transport_port_ssl` are set to the same
+value. The rules governing port assignment/encryption are summed up in
+the table below:
```

Comment:
This explanation - and matrix (!?) is extremely convoluted and no user will understand it. I didn't understand it :-( Isn't there better way to describe the behavior?
What was actually wrong with the old description I wrote?

Reply:
Well... because things did get convoluted. A lot. Changes introduced in #7992 brought some consensus (please see #7783 #7866 #7500), but I see no simple rule that would explain all that behavior.
Of course, I'm open to ideas of how to explain it better.